.github/workflows/nightly-build.yaml

```
@@ -78,55 +78,397 @@ jobs:
+  # `build` too, not just `prepare`: e2e-test must never install against
+  # images that don't exist yet or (pre-this-change) a floating :latest.
   # -------------------------------------------------------------------
   e2e-test:
     name: E2E validation
-    needs: prepare
+    needs: [prepare, build]
+    # TEMPORARY -- DO NOT MERGE: skip e2e-test to smoke-test the new
+    # prepare/build/publish job graph on this PR branch without waiting on
+    # E2E infra. Revert this hunk (and the matching publish `if:` below)
+    # before merging.
+    if: false
```

actionlint (1.7.12)

[error] constant expression "false" in condition. remove the if: section

(if-cond)